### PR TITLE
Hide enemy when they die, and fix spectacle points not carrying forward bug

### PR DESCRIPTION
**Pull request by emmettirl**

### CODE REVIEW

File: scripts/battle/GameState.cs [hunk 1]
```diff
@@ -99,10 +100,21 @@ public class GameState {
             ProcessCombo(matchingCombo);
         } else { ComboStackChangedCustomEvent?.Invoke(this, EventArgs.Empty); }
         DeselectDeadEnemy();
+        HideDeadEnemies();
```
Comment (VonJellybean) [inline scripts/battle/GameState.cs:103]: Move to the end of `PlaySelectedCard()`, after `Hand.DiscardCard()`

File: scripts/battle/GameState.cs [hunk 2]
```diff
@@ -70,6 +70,7 @@ public class GameState {
 
     public void StartTurn() {
         DeselectDeadEnemy();
+        HideDeadEnemies();
```
Comment (VonJellybean) [inline scripts/battle/GameState.cs:73]: Move to the end of `EndTurn()`

File: scripts/battle/GameState.cs (hunk 1 above)
Comment (emmettirl) [reply scripts/battle/GameState.cs:103]: Done

File: scripts/battle/GameState.cs (hunk 2 above)
Comment (emmettirl) [reply scripts/battle/GameState.cs:73]: These are here for the case of enemies knocking themselves out on their turn. Headbutt, blood and sand and all that. 
I don't think enemies use EndTurn(), so they'd remain selected at beginning of user turn if we move these.

Comment (VonJellybean) [reply scripts/battle/GameState.cs:73]: `EndTurn()` is what makes enemies take their turns, so we just need to run these after that and before the `StartTurn()` call.

Comment (emmettirl) [reply scripts/battle/GameState.cs]: True enough, was laser focused on the enemy turn and missed that. 
Made the move

**Review by VonJellybean — APPROVED** [review]

**Pull request merged**